hibernate-core/src/main/java/org/hibernate/type/descriptor/java/ByteTypeDescriptor.java

```diff
@@ -102,4 +106,58 @@ public int getDefaultSqlPrecision(Dialect dialect) {
 	public int getDefaultSqlScale() {
 		return 0;
 	}
+
+	@Override
+	public <X> Byte coerce(X value, CoercionContext coercionContext) {
+		if ( value == null ) {
+			return null;
+		}
+
+		if ( value instanceof Byte ) {
+			return (byte) value;
```

Comment:
not sure but probably it is better to cast the value to Byte

Reply:
It already checks for null previously